kratos/utilities/reduction_utilities.h

```diff
@@ -152,6 +152,37 @@ class MaxReduction
     }
 };
 
+//***********************************************************************************
+//***********************************************************************************
+//***********************************************************************************
+template<class TDataType, class TReturnType = TDataType>
+class AbsMaxReduction
+{
+public:
+    typedef TDataType   value_type;
+    typedef TReturnType return_type;
+
+    TReturnType mValue = std::numeric_limits<TReturnType>::lowest(); // deliberately making the member value public, to allow one to change it as needed
+
+    /// access to reduced value
+    TReturnType GetValue() const
+    {
+        return mValue;
+    }
+
+    /// NON-THREADSAFE (fast) value of reduction, to be used within a single thread
+    void LocalReduce(const TDataType value){
+        mValue = (std::abs(mValue) < std::abs(value)) ? value : mValue;
+    }
+
+    /// THREADSAFE (needs some sort of lock guard) reduction, to be used to sync threads
+    void ThreadSafeReduce(const AbsMaxReduction<TDataType, TReturnType>& rOther)
+    {
+        const std::lock_guard<LockObject> scope_lock(ParallelUtilities::GetGlobalLock());
+        mValue = (std::abs(mValue) < std::abs(rOther.mValue)) ? rOther.mValue : mValue;
```

Comment:
```suggestion
        LocalReduce(rOther.mValue);
```

this should work too right? Would be better since reduces code duplication

same for the other reduction

Reply:
Makes sense